lib/x.h

```diff
@@ -0,0 +1,43 @@
+// SPDX-FileCopyrightText: 2023-2025, Alejandro Colomar <alx@kernel.org>
+// SPDX-License-Identifier: BSD-3-Clause
+
+
+#ifndef SHADOW_INCLUDE_LIB_X_H_
+#define SHADOW_INCLUDE_LIB_X_H_
+
+
+#include "config.h"
+
+#include <errno.h>
+#include <stddef.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "shadowlog.h"
+
+
+#define x(p)                                                          \
+({                                                                    \
+	__auto_type  p_ = p;                                          \
+                                                                      \
+	exit_if_null(p_);                                             \
+	p_;                                                           \
+})
+
+
+inline void exit_if_null(void *p);
+
+
+inline void
+exit_if_null(void *p)
```

Comment:
The helper function exit_if_null() and the x() macro could benefit from more detailed documentation explaining.

Reply:
I thought exit_if_null() was quite clear from the name.  :)

You're right that x() needs a comment, though.